ci/ray_ci/doc/load_doc_cache.py

```diff
@@ -0,0 +1,109 @@
+import boto3
+import botocore
+import subprocess
+import tarfile
+import os
+import click
+
+S3_BUCKET = "ray-ci-results"
+DOC_BUILD_DIR_S3 = "doc_build"
+
+
+def find_latest_commit():
+    """Find latest commit that was pushed to origin/master that is also on local env."""
+    latest_commit = subprocess.check_output(
+        [
+            "git",
+            "log",
+            "--pretty=format:%H",
+            "--first-parent",
+            "origin/master",
+            "-n",
+            "1",
+        ]
+    ).decode("utf-8")
+    return latest_commit
+
+
+def fetch_cache_from_s3(commit, target_file_path):
+    """
+    Fetch doc cache archive from ray-ci-results S3 bucket
+
+    Args:
+        commit (str): The commit hash of the doc cache to fetch
+        target_file_path (str): The file path to save the doc cache archive
+    """
+    # Create an S3 client
+    s3 = boto3.client("s3")
+    s3_file_path = f"{DOC_BUILD_DIR_S3}/{commit}.tgz"
+    try:
+        print(f"Downloading {commit}.tgz from S3...")
+        s3.download_file(S3_BUCKET, s3_file_path, target_file_path)
+        print(f"Successfully downloaded {s3_file_path} to {target_file_path}")
+    except botocore.exceptions.ClientError as e:
+        print(f"Failed to download {s3_file_path} from S3: {str(e)}")
+        raise e
+
+
+def extract_cache(cache_path: str, doc_dir: str):
+    """
+    Extract the doc cache archive to overwrite the ray/doc directory
+
+    Args:
+        file_path (str): The file path of the doc cache archive
+    """
+    with tarfile.open(cache_path, "r:gz") as tar:
+        tar.extractall(doc_dir)
+    print(f"Extracted {cache_path} to {doc_dir}")
+
+
+def list_changed_and_added_files(ray_dir: str):
```

Comment:
this need to list changed files compared to the `latest_commit` from `find_latest_commit` i think; otherwise if i have a stack commit

commit_1
commit_2
commit_from_master_branch

this function might only return the diff between `commit_1` and `commit_2`?

Reply:
done! I added `git diff --name-only <latest_commit>`. Also latest master commit should be querying with parent as master instead of origin/master since master is really where the branch is created from.